### PR TITLE
fix(console): should not append slash in cors allowed uri

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

fix(console): should not append slash in cors allowed uri
Refer to: https://silverhand-io.slack.com/archives/C02A8G4HVAM/p1653978147830959

## Linear Issue Reference

N/A

## Testing

Test in the admin console.

### CODE REVIEW

File: packages/console/src/pages/ApplicationDetails/components/Settings.tsx [hunk 1]
```diff
@@ -105,7 +105,12 @@ const Settings = ({ oidcConfig }: Props) => {
           control={control}
           defaultValue={[]}
           rules={{
-            validate: createValidatorForRhf(uriPatternRules),
+            validate: createValidatorForRhf({
+              pattern: {
+                verify: uriValidator(false, true),
```
Comment (demonzoo) [inline packages/console/src/pages/ApplicationDetails/components/Settings.tsx:110]: what does the `false, true` mean? Would be better to give it a readable property name. E.g.
```
uriValidator({ allowBlank: true, validateAsUrlOrigin: true })
```

File: packages/phrases/src/locales/en.ts [hunk 2]
```diff
@@ -114,6 +114,7 @@ const translation = {
       empty: 'No Data',
       missing_total_number: 'Unable to find Total-Number in response headers.',
       invalid_uri_format: 'Invalid URI format',
+      invalid_origin_format: 'Invalid origin format',
```
Comment (demonzoo) [inline packages/phrases/src/locales/en.ts:117]: nit: Invalid URL origin format

File: packages/phrases/src/locales/zh-cn.ts [hunk 3]
```diff
@@ -114,6 +114,7 @@ const translation = {
       empty: '没有数据',
       missing_total_number: '无法从返回的头部信息中找到 Total-Number。',
       invalid_uri_format: '无效的 URI 格式',
+      invalid_origin_format: '无效的 origin 格式',
```
Comment (demonzoo) [inline packages/phrases/src/locales/zh-cn.ts:117]: nit: 无效的 URL origin 格式

File: packages/console/src/utilities/validator.ts [hunk 4]
```diff
@@ -14,3 +14,17 @@ export const uriValidator = (verifyBlank = true) => {
     return true;
   };
 };
+
+export const uriOriginValidator = ({ allowBlank = true }) => {
+  return (value: string) => {
+    if (!allowBlank && value.trim().length === 0) {
+      return true;
```
Comment (IceHe) [inline packages/console/src/utilities/validator.ts:21]: seems that if `allowBlank` is false, we should return false here?

File: packages/console/src/utilities/validator.ts [hunk 5]
```diff
@@ -1,6 +1,6 @@
-export const uriValidator = (verifyBlank = true) => {
+export const uriValidator = ({ allowBlank = true }) => {
   return (value: string) => {
-    if (!verifyBlank && value.trim().length === 0) {
+    if (!allowBlank && value.trim().length === 0) {
       return true;
     }
```
Comment (IceHe) [inline packages/console/src/utilities/validator.ts:5]: seems that if `allowBlank` is false, we should return false here?

File: packages/console/src/utilities/validator.ts [hunk 6]
```diff
@@ -14,3 +14,17 @@ export const uriValidator = (verifyBlank = true) => {
     return true;
   };
 };
+
+export const uriOriginValidator = ({ allowBlank = true }) => {
```
Comment (demonzoo) [inline packages/console/src/utilities/validator.ts:18]: origin validator should not allow blank

File: packages/console/src/utilities/validator.ts [hunk 7]
```diff
@@ -1,6 +1,6 @@
-export const uriValidator = (verifyBlank = true) => {
+export const uriValidator = ({ allowBlank = true }) => {
   return (value: string) => {
-    if (!verifyBlank && value.trim().length === 0) {
+    if (!allowBlank && value.trim().length === 0) {
```
Comment (demonzoo) [inline packages/console/src/utilities/validator.ts:3]: Also here, in what case would we allow blank for a URL input?

File: packages/console/src/utilities/validator.ts (hunk 4 above)
Comment (xiaoyijun) [reply packages/console/src/utilities/validator.ts:21]: this is a confusing naming, should rename back to `verifyBlank`

File: packages/console/src/utilities/validator.ts (hunk 6 above)
Comment (xiaoyijun) [reply packages/console/src/utilities/validator.ts]: this is a confusing naming, should rename back to verifyBlank, updated

File: packages/console/src/utilities/validator.ts (hunk 5 above)
Comment (xiaoyijun) [reply packages/console/src/utilities/validator.ts]: ditto

**Review by IceHe — APPROVED** [review]
Overall LGTM

**Pull request merged**